maptivist/src/main/webapp/filter.js

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Filters markers based on category
+ */
+
+function getMarkersByCategory() {
+    const category = document.getElementById("category");
+    fetch("/marker").then(response => response.json()).then((markers) => {
+        console.log(markers);
+        const markersToDisplay = filterByCategory(markers, category);
```

Comment:
Using a const here to escape the block scope is probably unsafe. I recommend using "await" instead:

const markers = await fetch("/marker").then(response -> response.json());
return filterByCategory(markers, category);

I think that will work, but may need some fixing.

https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await